ERCS/erc-7710.md

```diff
@@ -0,0 +1,177 @@
+necessitate ERC-4337.
+
+## Motivation
+
+The development of smart contracts on Ethereum has led to a diverse array of decentralized applications (dApps)
+that leverage composability to interact with one another in innovative ways. While current smart contracts are
+indeed capable of working together, enabling these interactions, especially in the realm of sharing capabilities
+or permissions, remains a tedious and often gas-expensive process, which lacks backwards compatibility.
+
+Currently, for a smart contract to interact with or utilize the functionality of another, it typically requires
+hardcoded permissions or the development of bespoke, intermediary contracts. This not only increases the complexity and
+development time but also results in higher deployment and execution gas costs. Moreover, the rigid nature of these
+interactions limits the ability to adapt to new requirements or to delegate specific, limited permissions in a dynamic
+manner.
+
+The proposed standard aims to simplify and standardize the process of delegation between contracts, reducing the
+operational complexity and gas costs associated with shared capabilities. By establishing a common framework for
+delegating permissions, we can streamline interactions within the Ethereum ecosystem, making contracts more flexible,
+cost-effective, and adaptable to the needs of diverse applications. This opens up new possibilities for collaboration
+and innovation, allowing dApps to leverage each other's strengths in a more seamless and efficient manner.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT
+RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Terms
+
+- A **Delegator** is a smart contract that can create a delegation. It implements the `ERC7710` interface, which
+  implements ERC-1271.
+- A **Delegation Manager** is a singleton smart contract that is responsible for validating delegation authority and
+  calling on the *Delegator* to execute an action. It implements the `ER7710Manager` interface.
+- A **delegation** is an authority given to another address to perform a specific action.
+- A **delegate** is a smart contract, smart contract account, or EOA that has authority to redeem a delegation.
+- A **redeemer** is a *delegate* that is using a delegation.
+
+### Overview
+
+#### Redeeming a Delegation
+
+When a delegate wishes to redeem a delegation, they call the `redeemDelegation` function on the Delegation Manager and
+pass in the Action they want to execute and the proof of authority (ie delegation) which they are executing on behalf
+of. The Delegation Manager then verifies the delegation's validity and, if valid, calls the privileged function on the
+Delegator which executes the specified capability on behalf of the Delegator.
+
+```mermaid
```

Comment:
Our renderer doesn't support mermaid. Please replace this with an image, preferably an SVG.

Reply:
have updated and added in an SVG in the assets folder for this.